### PR TITLE
adds interaction style to relationships

### PR DESCRIPTION
**Pull request by thedevelopnik**
Addresses issue #10 

Adds optional interaction style param (null by default) to `uses` and `usedBy` relationship methods. To keep from making a breaking change to the API, it comes after the technologies vararg parameter, so must be specified in passing parameters.

Added a test case for both uses and usedBy using the interactionStyle param. The existing test cases without it also serve as a test case for not passing the param, but happy to refactor this based on your feedback.

Thanks!

### CODE REVIEW

**Comment by thedevelopnik** [comment]
Here is the output from Travis on the failed check:

```
Installing oraclejdk8
$ export JAVA_HOME=~/oraclejdk8
$ export PATH="$JAVA_HOME/bin:$PATH"
$ ~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"
install-jdk.sh 2019-05-02
Expected feature release number in range of 9 to 13, but got: 8
The command "~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"" failed and exited with 3 during .
```

I looked in the Travis file and don't see anything obvious I could change to fix this. Any advice? I'll dig in a bit but I'm not very familiar with Travis and its particular issues as we use Circle.

**Comment by Exordian** [comment]
I just switched our project to `openjdk8` from the previously used `oraclejdk8` (on master). We don't rely on any oracle jdk specific features, and it seems that the oracle licensing changes bricked the travis ci build environment [1].

tl;dr: if you rebase your PR, the build should be fine

[1] https://travis-ci.community/t/install-of-oracle-jdk-8-failing/3038/2

**Comment by thedevelopnik** [comment]
It's alive!

**Comment by thedevelopnik** [comment]
@Exordian Is there any other work I should do to have this ready for merge?

**Comment by Exordian** [comment]
it's fine, thanks ! sorry for the delay

**Comment by thedevelopnik** [comment]
Thanks so much, for the merge and for getting the release out!
